keycloak/auth/auth.go

```diff
@@ -110,13 +115,24 @@ func (c *tokenSource) Token() (*oauth2.Token, error) {
 		return nil, fmt.Errorf("oauth2: cannot fetch token: %v\nResponse: %s", r.Status, body)
 	}
 
-	tk := &oauth2.Token{}
-
-	err = json.Unmarshal(body, tk)
+	var tokenRes struct {
+		RefreshToken string `json:"refresh_token"`
+		AccessToken  string `json:"access_token"`
+		TokenType    string `json:"token_type"`
+		IDToken      string `json:"id_token"`
+		ExpiresIn    int64  `json:"expires_in"` // relative seconds from now
+	}
 
+	err = json.Unmarshal(body, &tokenRes)
 	if err != nil {
 		return nil, err
 	}
 
+	tk := &oauth2.Token{
+		Expiry:       time.Now().Add(5 * time.Minute),
```

Comment:
Is 5 arbitrary or should it be based on ExpiresIn, given this can be configured in Keycloak?